Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def resolve_url(url)
+    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
+    m.nil? ? url : resolve_url(m.captures.first)
```

Comment:
Isn't it a concern that a(n https -> http) redirect could happen in a second or any subsequent redirect?